src/Rules/Functions/ArrayFilterStrictRule.php

```diff
@@ -0,0 +1,131 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\ArgumentsNormalizer;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Type;
+use PHPStan\Type\VerbosityLevel;
+use function count;
+use function sprintf;
+
+/**
+ * @implements Rule<FuncCall>
+ */
+class ArrayFilterStrictRule implements Rule
+{
+
+	/** @var ReflectionProvider */
+	private $reflectionProvider;
+
+	/** @var bool */
+	private $treatPhpDocTypesAsCertain;
+
+	/** @var bool */
+	private $reportMaybes;
+
+	public function __construct(
+		ReflectionProvider $reflectionProvider,
+		bool $treatPhpDocTypesAsCertain,
+		bool $reportMaybes
+	)
+	{
+		$this->reflectionProvider = $reflectionProvider;
+		$this->treatPhpDocTypesAsCertain = $treatPhpDocTypesAsCertain;
+		$this->reportMaybes = $reportMaybes;
+	}
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	/**
+	 * @param FuncCall $node
+	 * @return RuleError[] errors
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->name instanceof Name) {
+			return [];
+		}
+
+		if (!$this->reflectionProvider->hasFunction($node->name, $scope)) {
+			return [];
+		}
+
+		$functionReflection = $this->reflectionProvider->getFunction($node->name, $scope);
+
+		if ($functionReflection->getName() !== 'array_filter') {
+			return [];
+		}
+
+		$parametersAcceptor = ParametersAcceptorSelector::selectFromArgs(
+			$scope,
+			$node->getArgs(),
+			$functionReflection->getVariants(),
+			$functionReflection->getNamedArgumentsVariants()
+		);
+
+		$normalizedFuncCall = ArgumentsNormalizer::reorderFuncArguments($parametersAcceptor, $node);
+
+		if ($normalizedFuncCall === null) {
+			return [];
+		}
+
+		$args = $normalizedFuncCall->getArgs();
+		if (count($args) === 0) {
+			return [];
+		}
+
+		if (count($args) === 1) {
+			return [RuleErrorBuilder::message('Call to function array_filter() requires parameter #2 to be passed to avoid loose comparison semantics.')->build()];
+		}
+
+		$nativeCallbackType = $scope->getNativeType($args[1]->value);
+
+		if ($this->treatPhpDocTypesAsCertain) {
+			$callbackType = $scope->getType($args[1]->value);
+		} else {
+			$callbackType = $nativeCallbackType;
+		}
+
+		if ($this->isCallbackTypeNull($callbackType)) {
+			$message = 'Call to function array_filter() requires parameter #2 to be callable, %s given.';
```

Comment:
The message should be different so that people do not confuse it with this error https://phpstan.org/r/b11e20eb-740e-4e71-803e-7fab2a419512.

It should say `parameter #2 cannot be null to avoid loose comparison semantics` for example, or something like that.